src/upchunk.ts

```diff
@@ -305,37 +355,63 @@ export class UpChunk  {
     this.getChunk()
       .then(() => {
         this.attemptCount = this.attemptCount + 1;
-
+        this.lastChunkStart = new Date();
         return this.sendChunk()
       })
       .then((res) => {
         if (SUCCESSFUL_CHUNK_UPLOAD_CODES.includes(res.statusCode)) {
+          const lastChunkEnd = new Date();
+          const lastChunkInterval = (lastChunkEnd.getTime() - this.lastChunkStart.getTime()) / 1000;
+
+          this.chunkHistory.push({
+            chunk: this.chunkCount,
+            attempts: this.attemptCount,
+            chunkSize: this.chunk.size/1024,
+            timeInterval: lastChunkInterval,
+          });
+
           this.dispatch('chunkSuccess', {
             chunk: this.chunkCount,
+            chunkSize: this.chunkSize,
+            realChunkSize: this.chunk.size,
             attempts: this.attemptCount,
+            timeInterval: lastChunkInterval,
             response: res,
           });
 
           this.attemptCount = 0;
           this.chunkCount = this.chunkCount + 1;
+          this.nextChunkRangeStart = this.nextChunkRangeStart + this.chunkByteSize;
 
           if (this.chunkCount < this.totalChunks) {
+            // dynamic chunk sizing
+            if (this.dynamicChunkSize) {
+              if (lastChunkInterval < 10)
+              {
+                this.chunkSize = Math.min(this.chunkSize * 2,this.maxChunkSize);
+              } else if (lastChunkInterval > 30) {
+                this.chunkSize = Math.max(this.chunkSize / 2,this.minChunkSize);
+              }
+
+              // Now update the new chunkByteSize to the newly calculated chunk size
+              this.chunkByteSize = this.chunkSize * 1024;
+
+              // Re-estimate the total number of chunks, by adding the completed
+              // chunks to the remaining chunks
+              const remainingChunks = (this.file.size-this.nextChunkRangeStart)/this.chunkByteSize;
+              this.totalChunks = Math.ceil(this.chunkCount + remainingChunks);
+            }
             this.sendChunks();
           } else {
             this.success = true;
             this.dispatch('success');
           }
 
-          const chunkFraction = this.chunkCount / this.totalChunks;
-          const uploadedBytes = chunkFraction * this.file.size;
-
-          const percentProgress = (100 * uploadedBytes) / this.file.size;
-
-          this.dispatch('progress', percentProgress);
         } else if (TEMPORARY_ERROR_CODES.includes(res.statusCode)) {
           if (this.paused || this.offline) {
             return;
           }
+	  // console.warn('DEBUG: Caught a temporary error: %j',res);
```

Comment:
Same question as above

Reply:
Yes, those were intentional as well.  They were useful for debugging, but cause issues with the way the tests are written, so they're commented out on purpose.